.github/actions/pgx-init/action.yml

```diff
@@ -0,0 +1,32 @@
+name: 'pgx initialization'
+description: 'cargo pgx init'
+inputs:
+  working-directory:
+    description: 'The directory in which there is a pgx extension project'
+    required: true
+  pg_version:
+    description: 'The version of postgres to use'
+    required: false
+    default: 14
+outputs: {}
+runs:
+  using: "composite"
+  steps:
+      - name: setup pgx
+        shell: bash
+        run: cargo install --locked cargo-pgx
+        working-directory: ${{ inputs.working-directory }}
+      - name: pgx init
+        shell: bash
+        working-directory: ${{ inputs.working-directory }}
+        run: |
+          set -x
+          ls -ltrah /home/runner/.pgx || true
+          # pgx init can take a long time, and it re-compiles postgres even when there
+          # is a cached version. So, we can just check for the directory and
+          if find /home/runner/.pgx | grep $(awk -F "=" '/pg${{ inputs.pg_version }}/ {print $2}' /home/runner/.pgx/config.toml | tr -d '"'); then
+            echo "Already found pgx is initialized. Skipping 'cargo pgx init' command."
+          else
+            cargo pgx init --pg${{ inputs.pg_version }} download
+          fi
+          ls -ltrah /home/runner/.pgx
```

Comment:
Is this checking to see if the node running our github workflow is the same node already had `cargo pgx init` ran on it? I figured we'd eventually have pre-built images and run these tests on them. That way we'd never have to run `pgx init` if we dont want to.

Reply:
In a previous step it does a cache restore attempt, so it's checking for cache hit or miss.

Currently understanding if there is something I'm missing, for example postgres started by init that I didn't notice, i.e. if something is needed besides files that can be cached.

Also, for some reason when the tests run, I am having it fail sometimes in CI, and also slow. But on my laptop it's super fast and never fails, but I'm giving more tries. Looks like this when it fails. https://github.com/CoreDB-io/coredb/actions/runs/3753427332/jobs/6376678527

Reply:
I currently suspect it's related to resources and there is some subtle race condition when limited on resources, but not sure yet.

Reply:
the location of PGDATA is

`~/.pgx/data-14/`

that includes `postgresql.auto.conf`

Reply:
Hey I was off topic in this conversation, I had broken something with the cache restore of the target/ directory. Yes I do think docker image seems like a good way to go about it.

Reply:
Thanks for doing this. These tests were super slow. And even slower at first when we were running `init` on all pg 11-15.